src/main/ruby/core/io.rb

```diff
@@ -352,6 +356,17 @@ def inspect # :nodoc:
       format '#<IO::InternalBuffer:0x%x total=%p start=%p used=%p data=%p %s>',
              object_id, @total, @start, @used, @storage, content
     end
+
+    Truffle::Boot.delay do
+      if Truffle::Boot.get_option('polyglot.stdio')
+        def fill_read(io, count)
```

Comment:
This is of course not pretty but seems a decent workaround for now.
Maybe we can make it so polyglot reads (and writes) also return a native buffer to simplify and be consistent?
It won't be optimal for polyglot IO performance but probably good enough. cc @chrisseaton